.github/workflows/builNTest.yaml

```diff
@@ -0,0 +1,30 @@
+name: Build and Test
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+      - develop
+
+jobs:
+  build-and-test-on-all-OS:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: 21
+          distribution: temurin
+      - name: Build and Test with Gradle
+        run: |
+          ./gradlew
+          ./gradlew clean
+          ./gradlew build
+          ./gradlew run
```

Comment:
Test should be a separate step, I think.

Reply:
currently there are no tests being run. I can change the wording. It builds then run is the "full" which does not run transform at the moment, needs to be fixed eventually. Transform then picks up where run leaves off and creates the benchmarks. Reset may not be needed at all for this but I prefer to clean things up